MapboxSearch/sample/build.gradle

```diff
@@ -15,20 +15,42 @@ if (mapboxApiToken == null) {
     throw new Exception("API token is not specified.")
 }
 
-def enableSBS = true
+def apiType = null
+def apiTypeString = null
 
-if (project.hasProperty('ENABLE_SBS')) {
-    enableSBS = Boolean.valueOf(project.property('ENABLE_SBS'))
-} else if (System.getenv('ENABLE_SBS') != null) {
-    enableSBS = Boolean.valueOf(System.getenv('ENABLE_SBS'))
+if (project.hasProperty('API_TYPE')) {
+    apiTypeString = String.valueOf(project.property('API_TYPE'))
+} else if (System.getenv('API_TYPE') != null) {
+    apiTypeString = String.valueOf(System.getenv('ENABLE_SEARCH_BOX'))
+}
+
+if (apiTypeString == "SEARCH_BOX") {
+    apiType = "com.mapbox.search.ApiType.SEARCH_BOX"
+} else if (apiTypeString == "SBS") {
+    apiType = "com.mapbox.search.ApiType.SBS"
+} else if (apiTypeString == "GEOCODING") {
+    apiType = "com.mapbox.search.ApiType.GEOCODING"
+} else {
+    def enableSBS = false
+    if (project.hasProperty('ENABLE_SBS')) {
+        enableSBS = Boolean.valueOf(project.property('ENABLE_SBS'))
+    } else if (System.getenv('ENABLE_SBS') != null) {
+        enableSBS = Boolean.valueOf(System.getenv('ENABLE_SBS'))
+    }
+
+    if (enableSBS) {
+        apiType = "com.mapbox.search.ApiType.SBS"
+    } else {
+        apiType = "com.mapbox.search.ApiType.SEARCH_BOX"
+    }
```

Comment:
What do you think about adding a factory method to ApiType:
```
public enum class ApiType {

    /**
     * Geocoding API.
     */
    GEOCODING,

    /**
     * Single Box Search API.
     */
    @Reserved(Flags.SBS)
    SBS,

    /**
     * Search Box API
     */
    SEARCH_BOX;

    companion object {
        fun fromString(value: String): ApiType? {
            return when (value.toUpperCase()) {
                "GEOCODING" -> GEOCODING
                "SBS" -> SBS,
                "SEARCH_BOX" -> SEARCH_BOX,
                else -> null
            }
        }
    }
}
```

Reply:
Let's improve this as a follow-up task (I filed SSDK-864 for that)